pkg/prometheus/promcfg.go

```diff
@@ -116,6 +116,34 @@ func addTLStoYaml(cfg yaml.MapSlice, namespace string, tls *v1.TLSConfig) yaml.M
 	return cfg
 }
 
+func addSafeAuthorizationToYaml(cfg yaml.MapSlice, version semver.Version, assetStoreKey string, store *assets.Store, auth *v1.SafeAuthorization) yaml.MapSlice {
+	if auth == nil || !version.GTE(semver.MustParse("2.26.0")) {
+		return cfg
+	}
+	authCfg := yaml.MapSlice{}
+	if auth.Type == "" {
+		auth.Type = "Bearer"
+	}
+	authCfg = append(authCfg, yaml.MapItem{Key: "type", Value: strings.TrimSpace(auth.Type)})
```

Comment:
```suggestion
	if auth.Type != "" {
		authCfg = append(authCfg, yaml.MapItem{Key: "type", Value: strings.TrimSpace(auth.Type)})
	}
```

Reply:
Technically `Type` can be omitted and in that case defaults to `Bearer`. We could remove this explicit default and let prometheus handle it, I opted for explicitness here.